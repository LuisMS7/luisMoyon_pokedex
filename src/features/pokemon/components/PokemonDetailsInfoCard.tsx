import React from "react";
import { Text, StyleSheet, ScrollView } from "react-native";
import { perfectSize } from "@utils/perfect-size";
import { Pokemon } from "@pokemon/types/pokemon.type";
import { BOLD_WEIGHT, TEXT_BIG_SIZE, TEXT_TITLE_SIZE } from "@lib/constants";
import PokemonTypesInfo from "@pokemon/components/PokemonTypesInfo";
import PokemonSprites from "@pokemon/components/PokemonSprites";
import PokemonMovesInfo from "@pokemon/components/PokemonMovesInfo";

type PokemonDetailsInfoCardProps = {
	pokemon: Pokemon;
	typeColor: string;
};

const PokemonDetailsInfoCard = ({
	pokemon,
	typeColor,
}: PokemonDetailsInfoCardProps) => {
	return (
		<ScrollView
			style={styles.container}
			contentContainerStyle={styles.scrollView}
		>
			<PokemonTypesInfo pokemon={pokemon} />
			<Text style={{ ...styles.aboutTitle, color: typeColor }}>
				About
			</Text>
			<Text style={styles.weightKgText}>{pokemon.weight} kg</Text>
			<Text style={styles.weightKgText}>Weight</Text>
			<Text
				style={{
					...styles.aboutTitle,
					color: typeColor,
				}}
			>
				Sprites
			</Text>
			<PokemonSprites sprites={pokemon.sprites!} />
			<Text
				style={{
					...styles.aboutTitle,
					color: typeColor,
				}}
			>
				Moves
			</Text>
			<PokemonMovesInfo pokemon={pokemon} typeColor={typeColor} />
		</ScrollView>
	);
};

const styles = StyleSheet.create({
	container: {
		width: "95%",
		alignSelf: "center",
		flex: 1,
	},
	scrollView: {
		padding: perfectSize(20),
	},
	aboutTitle: {
		fontSize: TEXT_TITLE_SIZE,
		textAlign: "center",
		fontWeight: BOLD_WEIGHT,
		paddingVertical: 20,
	},
	weightKgText: {
		fontSize: TEXT_BIG_SIZE,
		textAlign: "center",
	},
});

export default PokemonDetailsInfoCard;
