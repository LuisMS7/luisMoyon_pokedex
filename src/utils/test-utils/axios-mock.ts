import client from "@lib/axios-instance";
import MockAdapter from "axios-mock-adapter";

export const mockClient = new MockAdapter(client);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const mockGetOnSuccess = (endpoint: string, mockResponse: any) => {
	mockClient.onGet(endpoint).reply(200, mockResponse);
};

export const mockGetOnFailure = (endpoint: string, statusCode: number) => {
	mockClient.onGet(endpoint).reply(statusCode);
};
