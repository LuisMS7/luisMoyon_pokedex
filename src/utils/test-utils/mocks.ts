import { Pokemon } from "@pokemon/types/pokemon.type";

export const mockPokemon: Pokemon = {
	id: 1,
	name: "mockPokemonName",
	// eslint-disable-next-line max-len
	img: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png",
	types: [{ type: { name: "normal", url: "" }, slot: 1 }],
	weight: 10,
	sprites: {
		back_default:
			// eslint-disable-next-line max-len
			"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png",
		back_female: "",
		back_shiny: "",
		back_shiny_female: "",
		front_default: "",
		front_female: "",
		front_shiny: "",
		front_shiny_female: "",
	},
	moves: [{ move: { name: "normal", url: "" }, version_group_details: [] }],
};
