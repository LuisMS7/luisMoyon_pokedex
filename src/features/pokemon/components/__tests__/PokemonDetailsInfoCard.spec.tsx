import React from "react";
import { render, screen } from "@testing-library/react-native";
import { mockPokemon } from "@utils/test-utils/mocks";
import PokemonDetailsInfoCard from "@pokemon/components/PokemonDetailsInfoCard";

describe("PokemonDetailsInfoCard", () => {
	it("should render correctly", () => {
		render(
			<PokemonDetailsInfoCard
				pokemon={mockPokemon}
				typeColor={"green"}
			/>,
		);

		expect(screen.toJSON()).toMatchSnapshot();
	});
});
