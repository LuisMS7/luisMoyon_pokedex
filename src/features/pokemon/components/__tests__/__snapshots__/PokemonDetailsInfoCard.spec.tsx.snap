// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetailsInfoCard should render correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "padding": 29.819916046431132,
    }
  }
  style={
    {
      "alignSelf": "center",
      "flex": 1,
      "width": "95%",
    }
  }
>
  <View>
    <View
      style={
        {
          "alignSelf": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "gap": 10,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#AAA67F",
            "borderColor": "transparent",
            "borderRadius": 1000,
            "borderWidth": 1,
            "justifyContent": "center",
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 17.89194962785868,
              "fontWeight": "800",
            }
          }
        >
          normal
        </Text>
      </View>
    </View>
    <Text
      style={
        {
          "color": "green",
          "fontSize": 35.78389925571736,
          "fontWeight": "800",
          "paddingVertical": 20,
          "textAlign": "center",
        }
      }
    >
      About
    </Text>
    <Text
      style={
        {
          "fontSize": 23.855932837144906,
          "textAlign": "center",
        }
      }
    >
      10
       kg
    </Text>
    <Text
      style={
        {
          "fontSize": 23.855932837144906,
          "textAlign": "center",
        }
      }
    >
      Weight
    </Text>
    <Text
      style={
        {
          "color": "green",
          "fontSize": 35.78389925571736,
          "fontWeight": "800",
          "paddingVertical": 20,
          "textAlign": "center",
        }
      }
    >
      Sprites
    </Text>
    <RCTScrollView
      contentContainerStyle={
        {
          "alignSelf": "center",
          "flexDirection": "row",
          "gap": 14.909958023215566,
        }
      }
      horizontal={true}
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 1000,
              "height": 119.27966418572453,
              "justifyContent": "center",
              "width": 119.27966418572453,
            }
          }
        >
          <Image
            resizeMode="cover"
            source={
              {
                "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png",
              }
            }
            style={
              {
                "height": 89.4597481392934,
                "width": 89.4597481392934,
              }
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 1000,
              "height": 119.27966418572453,
              "justifyContent": "center",
              "width": 119.27966418572453,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 1000,
              "height": 119.27966418572453,
              "justifyContent": "center",
              "width": 119.27966418572453,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 1000,
              "height": 119.27966418572453,
              "justifyContent": "center",
              "width": 119.27966418572453,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 1000,
              "height": 119.27966418572453,
              "justifyContent": "center",
              "width": 119.27966418572453,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 1000,
              "height": 119.27966418572453,
              "justifyContent": "center",
              "width": 119.27966418572453,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 1000,
              "height": 119.27966418572453,
              "justifyContent": "center",
              "width": 119.27966418572453,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 1000,
              "height": 119.27966418572453,
              "justifyContent": "center",
              "width": 119.27966418572453,
            }
          }
        />
      </View>
    </RCTScrollView>
    <Text
      style={
        {
          "color": "green",
          "fontSize": 35.78389925571736,
          "fontWeight": "800",
          "paddingVertical": 20,
          "textAlign": "center",
        }
      }
    >
      Moves
    </Text>
    <View
      style={
        {
          "alignSelf": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "gap": 10,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "green",
            "borderRadius": 1000,
            "borderWidth": 1,
            "justifyContent": "center",
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
      >
        <Text
          style={
            {
              "color": "black",
              "fontSize": 17.89194962785868,
              "fontWeight": "400",
            }
          }
        >
          normal
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
